Src/FluentAssertions/DataRowCollectionAssertionExtensions.cs

```diff
@@ -0,0 +1,827 @@
+        /// </summary>
+        /// <param name="otherCollection">The other <see cref="DataRowCollection"/> with the unexpected number of elements</param>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+        /// </param>
+        public static AndConstraint<GenericCollectionAssertions<DataRow>> NotHaveSameCount(this GenericCollectionAssertions<DataRow> assertion, DataRowCollection otherCollection, string because = "",
+            params object[] becauseArgs)
+        {
+            Guard.ThrowIfArgumentIsNull(otherCollection, nameof(otherCollection), "Cannot verify count against a <null> collection.");
+
+            Execute.Assertion
+                .BecauseOf(because, becauseArgs)
+                .WithExpectation("Expected {context:collection} to not have ")
+                .Given(() => assertion.Subject)
+                .ForCondition(subject => subject is not null)
+                .FailWith("the same count as {0}{reason}, but found <null>.", otherCollection)
+                .Then
+                .Given((subject) => (actual: subject.Count(), expected: otherCollection.Count))
+                .ForCondition(count => count.actual != count.expected)
+                .FailWith("{0} row(s){reason}, but found {1}.", count => count.expected, count => count.actual)
+                .Then
+                .ClearExpectation();
+
+            return new AndConstraint<GenericCollectionAssertions<DataRow>>(assertion);
+        }
+
+        /// <summary>
+        /// Asserts that a <see cref="DataRowCollection"/> contains at least one <see cref="DataRow"/> equivalent to another <see cref="DataRow"/>.
+        /// </summary>
+        /// <remarks>
+        /// See the documentation for <see cref="DataRowAssertions{TDataRow}"/> for information about when <see cref="DataRow"/>
+        /// objects are considered equivalent.
+        /// </remarks>
+        /// <param name="expectation">The expected <see cref="DataRow"/>.</param>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <paramref name="because"/>.
+        /// </param>
+        public static AndConstraint<GenericCollectionAssertions<DataRow>> ContainEquivalentOf(this GenericCollectionAssertions<DataRow> assertion, DataRow expectation, string because = "",
```

Comment:
🤔 Why do you need these methods at all? If so, cover them with tests, including all the guards and edge cases. Same for all `ContainEquivalentOf` and `NotContainEquivalentOf` overloads.